src/TriggerBinding/SqlTriggerListener.cs

```diff
@@ -463,6 +470,124 @@ PRIMARY KEY ({primaryKeys})
             }
         }
 
+        async Task<ScaleMetrics> IScaleMonitor.GetMetricsAsync()
+        {
+            return await this.GetMetricsAsync();
+        }
+
+        public async Task<SqlTriggerMetrics> GetMetricsAsync()
+        {
+            Debug.Assert(!(this._changeMonitor is null));
+
+            return new SqlTriggerMetrics
+            {
+                UnprocessedChangeCount = await this._changeMonitor.GetUnprocessedChangeCountAsync(),
+                Timestamp = DateTime.UtcNow,
+            };
+        }
+
+        ScaleStatus IScaleMonitor.GetScaleStatus(ScaleStatusContext context)
+        {
+            return this.GetScaleStatusWithTelemetry(context.WorkerCount, context.Metrics?.Cast<SqlTriggerMetrics>().ToArray());
+        }
+
+        public ScaleStatus GetScaleStatus(ScaleStatusContext<SqlTriggerMetrics> context)
+        {
+            return this.GetScaleStatusWithTelemetry(context.WorkerCount, context.Metrics?.ToArray());
+        }
+
+        private ScaleStatus GetScaleStatusWithTelemetry(int workerCount, SqlTriggerMetrics[] metrics)
+        {
+            var measures = new Dictionary<TelemetryMeasureName, double>
+            {
+                [TelemetryMeasureName.WorkerCount] = workerCount,
+                [TelemetryMeasureName.UnprocessedChangeCount] = (metrics is null || metrics.Length == 0) ? -1L : metrics[metrics.Length - 1].UnprocessedChangeCount,
```

Comment:
This doesn't seem super useful to have without having all the metrics being considered. I'd just have this be a stringified array (that we can process on the server side if needed) of the counts that are being considered - e.g. `[10, 20, 30, 40, 50]`

Reply:
The value needs to be of type double. Let me know if you have any suggestion.

Reply:
Just have it be a property instead - it doesn't really make a huge difference which bag it ends up in since we manually process all our events anyways for our reporting. 

I'd also simplify this and just have a single event `GetScaleStatus` that's sent at the end. I don't really remember the reason why we have a bunch of start/end events for the other ones but it generally isn't that useful and just makes combining the data to be more tedious. 

And then also wrap the call to get the scale status in a try/catch and track error event if it fails.